File: 1249-minimum-remove-to-make-valid-parentheses/1249-minimum-remove-to-make-valid-parentheses.cpp
class Solution {
public:
    string minRemoveToMakeValid(string s) {
        int open = 0;
        string str = "";
        for(auto c: s){
            if(c!='(' && c!=')'){
                str += c;
                continue;
            }
            else if(c=='('){
                open++;
                str += c;
            }
            else{
                if(open>0){
                    open--;
                    str += c;
                }
            }
        }
        cout<<str;
        string ans = "";
        if(open){
            for(int i=str.size()-1;i>=0;i--){
                if(open && str[i]=='('){
                    open--;
                    continue;
                }
                else{
                    ans += str[i];
                }
            }
            reverse(ans.begin(),ans.end());
        }
        else{
            ans = str;
        }
        return ans;
    }
};

File: 1249-minimum-remove-to-make-valid-parentheses/1249-minimum-remove-to-make-valid-parentheses.py
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        stack = []
        for i in range(len(s)):
            if s[i] in '()':
                if s[i] == ')' and len(stack) and s[stack[-1]]  == '(':
                    stack.pop()
                else:
                    stack.append(i)
        for i in stack:
            s = s[:i] + '#' +s[i + 1:]
        s = s.replace('#','')
        return s